code/modules/organs/prosthetics/_prosthetics.dm

```diff
@@ -0,0 +1,196 @@
+/*
+  A system for easily and quickly removing your own bodyparts, with a view towards
+  swapping them out for new ones, or just doing it as a party trick to horrify an
+  audience. Current implementation only supports robolimbs and uses a modular_bodypart
+  value on the manufacturer datum, but I have tried to keep it generic for future work.
+  PS. jesus christ this was meant to be a half an hour port
+*/
+
+// External organ procs:
+// Does this bodypart count as a modular limb, and if so, what kind?
+/obj/item/organ/external/proc/get_modular_limb_category()
+	. = MODULAR_BODYPART_INVALID
+	if(BP_IS_PROSTHETIC(src) && model)
+		var/decl/prosthetics_manufacturer/manufacturer = decls_repository.get_decl(model)
+		if(!isnull(manufacturer?.modular_bodyparts))
+			. = manufacturer.modular_bodyparts
```

Comment:
Confusing var name; made me look up whether this was a typo.

Reply:
Renamed as discussed.